src/components/network/line-layer.js

```diff
@@ -50,44 +49,124 @@ function getArrowDirection(p) {
+    if (zone === LineLoadingZone.UNKNOWN) {
+        return [128, 128, 128]; // grey
+    } else if (zone === LineLoadingZone.SAFE) {
+        return [107, 178, 40]; // green
+    } else if (zone === LineLoadingZone.WARNING) {
+        return [210, 179, 63]; // yellow
+    } else if (zone === LineLoadingZone.OVERLOAD) {
+        return [255, 0, 0]; // red
+    } else {
+        throw new Error('Unsupported line loading zone: ' + zone);
+    }
+}
+
+function getLineColor(line, nominalVoltageColor, props) {
     if (props.lineFlowColorMode === LineFlowColorMode.NOMINAL_VOLTAGE) {
         if (isDisconnected(line)) {
             return props.disconnectedLineColor;
         } else {
-            return color;
+            return nominalVoltageColor;
         }
     } else if (props.lineFlowColorMode === LineFlowColorMode.OVERLOADS) {
-        let limits = [line.permanentLimit1, line.permanentLimit2];
-        let intensities = [line.i1, line.i2];
-        let iColor1, iColor2;
-        let iColor = [iColor1, iColor2];
-
-        for (let i = 0; i < 2; ++i) {
-            if (
-                limits[i] === undefined ||
-                intensities[i] === undefined ||
-                intensities[i] === 0
-            ) {
-                iColor[i] = 0;
-            } else {
-                let threshold =
-                    (props.lineFlowAlertThreshold * limits[i]) / 100;
-                if (intensities[i] > 0 && intensities[i] < threshold) {
-                    iColor[i] = 1;
-                } else if (
-                    intensities[i] >= threshold &&
-                    intensities[i] < limits[i]
-                ) {
-                    iColor[i] = 2;
-                } else {
-                    iColor[i] = 3;
-                }
-            }
-        }
-        return getOverLoadLineColor(Math.max(iColor[0], iColor[1]));
+        const zone = getLineLoadingZone(line, props.lineFlowAlertThreshold);
+        return getLineLoadingZoneColor(zone);
     } else {
-        return color;
+        return nominalVoltageColor;
+    }
+}
+
+export const ArrowSpeed = {
+    STOPPED: 0,
+    SLOW: 1,
+    MEDIUM: 2,
+    FAST: 3,
+    CRAZY: 4,
+};
+
+function getArrowSpeedOfSide(limit, intensity) {
+    if (limit === undefined || intensity === undefined || intensity === 0) {
+        return ArrowSpeed.STOPPED;
+    } else {
+        if (intensity > 0 && intensity < limit / 3) {
+            return ArrowSpeed.SLOW;
+        } else if (intensity >= limit / 3 && intensity < (limit * 2) / 3) {
+            return ArrowSpeed.MEDIUM;
+        } else if (intensity >= (limit * 2) / 3 && intensity < limit) {
+            return ArrowSpeed.FAST;
+        } else {
+            // > limit
+            return ArrowSpeed.CRAZY;
+        }
+    }
+}
+
+function getArrowSpeed(line, lineFlowAlertThreshold) {
+    const speed1 = getArrowSpeedOfSide(
+        line.permanentLimit1,
+        line.i1,
+        lineFlowAlertThreshold
```

Comment:
third parameter lineFlowAlertThreshold not used in getArrowSpeedOfSide function
you can also remove it as second parameter of getArrowSpeed function

Reply:
Fixed